ec2-spot-sagemaker-managed-spot-training/sagemaker-deepar.ipynb

```diff
@@ -0,0 +1,671 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "time_series_training = []\n",
+    "for ts in time_series:\n",
+    "    time_series_training.append(ts[:-prediction_length])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "time_series[0].plot(label='test')\n",
+    "time_series_training[0].plot(label='train', ls=':')\n",
+    "plt.legend()\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following utility functions convert `pandas.Series` objects into the appropriate JSON strings that DeepAR can consume. We will use these to write the data to S3."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def series_to_obj(ts, cat=None):\n",
+    "    obj = {\"start\": str(ts.index[0]), \"target\": list(ts)}\n",
+    "    if cat is not None:\n",
+    "        obj[\"cat\"] = cat\n",
+    "    return obj\n",
+    "\n",
+    "def series_to_jsonline(ts, cat=None):\n",
+    "    return json.dumps(series_to_obj(ts, cat))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "encoding = \"utf-8\"\n",
+    "s3filesystem = s3fs.S3FileSystem()\n",
+    "\n",
+    "print(s3_data_path)\n",
+    "\n",
+    "with s3filesystem.open(s3_data_path + \"/train/train.json\", 'wb') as fp:\n",
+    "    for ts in time_series_training:\n",
+    "        fp.write(series_to_jsonline(ts).encode(encoding))\n",
+    "        fp.write('\\n'.encode(encoding))\n",
+    "\n",
+    "with s3filesystem.open(s3_data_path + \"/test/test.json\", 'wb') as fp:\n",
+    "    for ts in time_series:\n",
+    "        fp.write(series_to_jsonline(ts).encode(encoding))\n",
+    "        fp.write('\\n'.encode(encoding))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Train a model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now define the estimator that will launch the training job."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "estimator = sagemaker.estimator.Estimator(\n",
+    "    sagemaker_session=sagemaker_session,\n",
+    "    image_name=image_name,\n",
+    "    role=role,\n",
+    "    train_instance_count=1,\n",
+    "    train_instance_type='ml.c4.xlarge',\n",
+    "    train_use_spot_instances=True,\n",
```

Comment:
I'd suggest adding a markdown section either before or after this block, Where we note and highlight what has changed and what is required to enable the spot use; Ideally with links to SDK documentation.